IPCLib/Socket.cpp

```diff
@@ -20,40 +43,48 @@ ReceivingThread::ReceivingThread(const std::function<void(bool*)>& p_receiveData
 /// @return Whether the thread received a message
 bool ReceivingThread::HasReceivedMessage() const
 {
-    return m_received;
+    return GET_RECEIVED();
+}
+
+/// @brief  Check whether the tread threw an error
+/// @return Whether the thread threw an error
+bool ReceivingThread::HasError() const
+{
+    if (!GET_ERROR()) return false;
+    return true;
 }
 
-/// @brief  Checks whether the thread started receiving
-/// @return Whether the thread started receiving
+/// @brief  Check whether the thread has started receiving
+/// @return Whether the thread has started receiving
 bool ReceivingThread::StartedReceiving() const
 {
-    return m_startedReceiving;
+    return GET_STARTED_RECEIVING() || GET_RECEIVED();
```

Comment:
Started receiving will be true if GET_RECEIVED() is true? Is this what you need? `GET_STARTED_RECEIVING() || GET_RECEIVING()` makes more sense to me

Reply:
no the condition should be true if it started or has received a message, as both state that the worker thread started running but we do not know at what tume we are checking